ImagePicker/ImagePickerAssetModel.swift

```diff
@@ -1,89 +1,67 @@
-//
-//  ImagePickerAssetModel.swift
-//  ImagePicker
-//
-//  Created by Peter Stajger on 12/09/2017.
-//  Copyright © 2017 Inloop. All rights reserved.
-//
+// Copyright © 2018 INLOOPX. All rights reserved.
 
-import Foundation
 import Photos
 
-///
 /// Model that is used when accessing an caching PHAsset objects
-///
+
 final class ImagePickerAssetModel {
-    
-    deinit {
-        log("deinit: \(String(describing: self))")
-    }
-    
     var fetchResult: PHFetchResult<PHAsset>! {
         set { userDefinedFetchResult = newValue }
         get { return userDefinedFetchResult ?? defaultFetchResult }
     }
     
+    // Will be use for caching
+    var previousPreheatRect = CGRect.zero
     lazy var imageManager = PHCachingImageManager()
     var thumbnailSize: CGSize?
     
     /// Tryies to access smart album .smartAlbumUserLibrary that should be `Camera Roll` and uses just fetchAssets as fallback
     private lazy var defaultFetchResult: PHFetchResult<PHAsset> = {
-        
         let assetsOptions = PHFetchOptions()
         assetsOptions.sortDescriptors = [NSSortDescriptor(key: "creationDate", ascending: false)]
         assetsOptions.fetchLimit = 1000
         
         let collections = PHAssetCollection.fetchAssetCollections(with: .smartAlbum, subtype: .smartAlbumUserLibrary, options: nil)
         if let cameraRoll = collections.firstObject {
             return PHAsset.fetchAssets(in: cameraRoll, options: assetsOptions)
-        }
-        else {
+        } else {
             return PHAsset.fetchAssets(with: assetsOptions)
         }
     }()
-    
     private var userDefinedFetchResult: PHFetchResult<PHAsset>?
     
-    //will be use for caching
-    var previousPreheatRect = CGRect.zero
+    deinit {
+        log("deinit: \(String(describing: self))")
+    }
     
     func updateCachedAssets(collectionView: UICollectionView) {
-        
         // Paradoxly, using this precaching the scrolling of images is more laggy than if there is no precaching
-        
         guard let thumbnailSize = thumbnailSize else {
             return log("asset model: update cache assets - thumbnail size is nil")
         }
-        
+    
         guard let layout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout else {
             return log("asset model: update cache assets - collection view layout is not flow layout")
         }
         
         // The preheat window is twice the height of the visible rect.
         let visibleRect = CGRect(origin: collectionView.contentOffset, size: collectionView.bounds.size)
-        
         var preheatRect: CGRect
         
         switch layout.scrollDirection {
         case .vertical:
```

Comment:
both cases are pretty much the same - can you extract it to function and unify it?